test/python/circuit/test_controlled_gate.py

```diff
@@ -207,6 +207,25 @@ def test_single_controlled_composite_gate(self):
         ref_mat = Operator(qc).data
         self.assertTrue(matrix_equal(cop_mat, ref_mat, ignore_phase=True))
 
+    def test_multi_control_z(self):
```

Comment:
Since your change affects the gates used in the decomposition, the test should also verify that. This test would also pass with the old version, with the inefficient MCZ decomposition. Instead I think we should test something like
```python
self.assertEqual(ctr_circ.decompose(), ref_circ)
```

Reply:
Yes thanks a lot. I committed a new version of the test.